algorithms/CPlusPlus/Trees/print-all-nodes-at-a-level.cpp

```diff
@@ -0,0 +1,80 @@
+/* Description - We have to print all nodes at a level 'k' of the tree.
+
+For example- If we are given the following tree, the nodes at level 1 are 1, nodes at level 2 are 2,3, nodes at level 3 are 7, 9 and nodes at level 4 are 21 and 11.
+(Note- The level starts from 1, i.e root is at level 1)
+
+            1          level 1
+          /   \
+         2     3       level 2
+      	      /  \    
+             7    9    level 3
+	    /      \
+	   21      11  level 4
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+typedef struct Node
+{
+	int data;
+	struct Node* left;
+	struct Node* right;
+	
+	Node(int val)
+	{
+		data= val;
+		left=right=NULL;
+	}
+}node;
+
+//function for level order traversal
+void levelorder(Node* root, int k)
```

Comment:
Maybe in C++ this is valid(not really sure) but in C it will throw in an `unkown type name` error, since the name that comes after the brackets is the new typename which is `node` and not `Node` (there exist only `struct Node` not `Node` alone). It would be correct to write `struct Node` here but that would defeat the whole point of `typedef` . It's better to stick with the new typename(`node`) you defined to avoid any confusion.

Also change it everywhere in you code.
```suggestion
void levelorder(node* root, int k)
```